src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/EditingTargetCodeExecution.java

```diff
@@ -158,6 +161,44 @@ public void executeRange(Range range)
       executeRange(range, null, false);
    }
    
+   public void sendSelectionToTerminal(boolean terminalExecuteWhenNotFocused)
+   {
+      if (terminalExecuteWhenNotFocused && !docDisplay_.isFocused())
+      {
+         // the current editor may be in another window; if one of our source
+         // windows was last focused, use that one instead
+         SourceWindowManager manager = 
```

Comment:
Mind you that's pretty iffy, as the main menu state depends on the window with focus, not the last satellite editor you had selected!